lib/simp/cli/commands/bootstrap.rb

```diff
@@ -155,91 +91,133 @@ def self.run(args = [])
 
     # Print intro
     system('clear')
-    puts
-    puts "*** Starting SIMP Bootstrap ***"
-    puts "   If this runs quickly, something wrong happened. To debug the problem,"
-    puts "   run 'puppet agent --test' by hand or read the log. The log can be found"
-    puts "   at '#{@logfile.path}'."
-    puts
+    say ("<%= color('=== Starting SIMP Bootstrap ===', :yellow) %>")
+
+    # Set an interrupt trap if safe mode is enabled
+    say "> The log can be found at '#{@logfile.path}'\n"
+    if not @unsafe
+      Signal.trap("INT") { say "<%= color('Safe mode enabled, ignoring interrupt', :magenta) %>"}
```

Comment:
SIGHUP is a good addition. Not sure about SIGUSR1 or 2. It's best bootstrap not be interrupted at all if you want a sane system. If you're debugging or don't care, you can run with the --unsafe flag, then you can interrupt all you want.